numpy/_core/memmap.py

```diff
@@ -250,7 +250,7 @@ def __new__(subtype, filename, dtype=uint8, mode='r+', offset=0,
                 size = bytes // _dbytes
                 shape = (size,)
             else:
-                if type(shape) not in (tuple, list):
+                if not isinstance(shape, (tuple, list)):
```

Comment:
https://docs.astral.sh/ruff/rules/non-pep604-isinstance/
 
```suggestion
                if not isinstance(shape, tuple | list):
```

Reply:
That would be rule [UP038](https://docs.astral.sh/ruff/rules/non-pep604-isinstance/#non-pep604-isinstance-up038) which is deprecated. See https://github.com/astral-sh/ruff/issues/7871.